draft-ietf-quic-transport.md

```diff
@@ -3139,7 +3139,8 @@ attacks that successfully guess the IP address 3-tuple and reduce the PMTU to a
 bandwidth-inefficient value.
 
 An endpoint MUST ignore an ICMP message that claims the PMTU has decreased below
-1280 bytes.
+1280 bytes.  And endpoint MUST ignore an ICMP message that attempts to increase
+PMTU, unless the ICMP message was received in response to a PMTU probe packet.
```

Comment:
Not sure about the "unless" part here.

Say I expect PMTU to be 1280.  Then I send a probe for 2048.  I get back an ICMP PTB that says 1500.  This would allow the PMTU to be set to 1500, but I think that you want to probe at 1500 before increasing anything.

Thus, I would instead say that an ICMP message that claims a larger PMTU than the confirmed PMTU MUST be confirmed (by sending a packet at that size and having it acknowledged) before the PMTU can be increased.

Reply:
See the note above where I was wondering if people would read it the way you did.

The "unless" part only cancels "ignore the message" instruction.  It does not recommend using the message's MTU as PMTU w/o probing it first.

I did not want to get into PLDPMTUD probe size selection details, but, yet, you were confused by the text, so I will add a few more words...

Reply:
It seems this thread is re-inventing machinery to do path probing.

I am really unsure why this would propose  a different approach than the TSVWG draft? - It would be useful to have a starting point to understand why QUIC would behave in a different way to other datagram transports.

Reply:
Wouldn't the best thing here be to tweak the next paragraph to say that any reactions to ICMP messages should be validated and DPLPMTUD do specify how to do such validation both for increasing and decreasing the PMTU?

Reply:
@gloinul, DPLPMTUD is not a requirement for QUIC.

This text was to apply to all implementations -- ones performing DPLPMTUD or traditional PMTUD or something else.  It is possible that we just reject this PR and assume people will follow directions specified in whatever PMTUD scheme they are implementing.